### DISCUSSION TITLE
Issue with [@types/minimatch] for minimatch 6.1.2

### DISCUSSION THREAD

**Discussion by FloLey** [DefinitelyTyped/DefinitelyTyped#63967] [Issues with a @types package]
I'm getting the following errors on a project that just updated minimatch from 5.1.2 to 6.1.2. Is there something I can do to fix this or should I revert to version 5.1.2?

node_modules/@types/glob/index.d.ts(29,42): error TS2694: Namespace '"/home/jenkins/workspace/bot_npm_and_yarn_minimatch-6.1.2/node_modules/minimatch/dist/cjs/index"' has no exported member 'IOptions'.

node_modules/@types/glob/index.d.ts(74,30): error TS2724: '"/home/jenkins/workspace/bot_npm_and_yarn_minimatch-6.1.2/node_modules/minimatch/dist/cjs/index"' has no exported member named 'IMinimatch'. Did you mean 'Minimatch'?

Thanks in advance

[a comment by an automated account is not shown]

**Comment by BendingBender** [comment]
@FloLey  I made a PR to remove the types from DT. You can simply uninstall the `@types/minimatch` package from your project to fix your problem because `minimatch` since v6 is written in TS and provides its own types.

see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63975

**Reply by FloLey** [reply]
Thanks a lot! I see the PR has been merged but I don't find where I could see if this has been released or not. Could you tell me? :)

**Reply by BendingBender** [reply]
There is nothing to release, `minimatch@6` provides its own types, you simply don't need the `@types/minimatch` package any more. Remove it via
```
npm uninstall @types/minimatch
```
or
```
yarn remove @types/minimatch
```

**Reply by FloLey** [reply]
I tried that but I still get:
```
node_modules/@types/glob/index.d.ts(29,42): error TS2694: Namespace '"/home/jenkins/workspace/t_npm_and_yarn_minimatch-6.1.6_2/node_modules/minimatch/dist/cjs/index"' has no exported member 'IOptions'.

node_modules/@types/glob/index.d.ts(74,30): error TS2724: '"/home/jenkins/workspace/t_npm_and_yarn_minimatch-6.1.6_2/node_modules/minimatch/dist/cjs/index"' has no exported member named 'IMinimatch'. Did you mean 'Minimatch'?
```

**Reply by BendingBender** [reply]
Hm, sadly there's not much we can do about it. The problem is that `@types/glob` has a dependency on `@types/minimatch@5`. But you have `minimatch@6` installed. When Typescript sees a package that provides its own types (like `minimatch@6`) and a type definition package (like `@types/minimatch@5` installed by `@types/glob`) it will always prefer the types coming from the actual package. The problem is now that the types from `minimatch@6` are incompatible with the ones required by `@types/glob`, hence the error you see.

The only way forward to fix this situation is to update the `glob` package to use `minimatch@6` and then update the type definitions (`@types/glob`) to use the types from `minimatch@6` instead of `@types/minimatch@5`. Another solution is to stick with `minimatch@5` until `glob` and `@types/glob` are updated accordingly. You can try to make a PR to the `glob` repo or ask there to update the `minimatch` dependency. Yet another workaround would be to enable the `skipLibCheck` compiler option, but it can have a few side effects.

**Reply by BendingBender** [reply]
This seems to be already being worked on, see https://github.com/isaacs/node-glob/issues/490 and https://github.com/isaacs/node-glob/pull/489.
